check_api/src/main/java/com/google/errorprone/util/SourceVersion.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 The Error Prone Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.errorprone.util;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.sun.tools.javac.code.Source;
+import com.sun.tools.javac.code.Source.Feature;
+import com.sun.tools.javac.util.Context;
+import java.util.Arrays;
+
+/**
+ * JDK source version utilities.
+ *
+ * @see RuntimeVersion
+ */
+public final class SourceVersion {
+  private static final ImmutableMap<String, Feature> KNOWN_FEATURES =
+      Maps.uniqueIndex(Arrays.asList(Feature.values()), Enum::name);
```

Comment:
The string-based lookup is necessary because older runtimes won't recognize newer enum values. A bit fragile, but no more fragile than what's usual for Error Prone.

Reply:
Thanks, I didn't consider that when making the suggestion. This looks like a good way to deal with that.

One more suggestion would be to keep the current API, but avoid relying on the `Feature` enum, e.g.:

```java
public static boolean supportsSwitchExpressions(Context context) {
  return Source.instance(state.context).compareTo(Source.lookup("14")) >= 0;
}
```

That means we can't use javac's knowledge about which features were used in which versions, but it also avoids depending on the string names of the `Feature` constants, and they're an internal API that could change (e.g. constants will probably be removed as javac drops support for old version features).